### PR TITLE
[Documentation] bot.py doc fixes

### PR DESCRIPTION
**Pull request by harshil21**
All docs obtained from official Bot API docs.

### CODE REVIEW

**Comment by harshil21** [comment]
There is also documentation related to API 4.7 in there

**Comment by Bibo-Joshi** [comment]
Wow, thanks! Will have to merge after #1858 though, if there is API 4.7 docs in there.
Also, flake8 is unhappy:

```
telegram/bot.py:506:100: E501 line too long (100 > 99 characters)
```

If you install the pre-commit hooks, git will check flake8 before commiting ;) see our [contrib-guide](https://github.com/python-telegram-bot/python-telegram-bot/blob/master/.github/CONTRIBUTING.rst#setting-things-up)

**Comment by harshil21** [comment]
Yea I didn't add all docs related to 4.7 since the new methods aren't there in `bot.py` yet.
Also I use GH desktop instead of git so that's why I can't run the tests before commiting :(

**Comment by Bibo-Joshi** [comment]
> Yea I didn't add all docs related to 4.7 since the new methods aren't there in `bot.py` yet.

I guessed so ;) But we'll release the API 4.7 changes soon.

> Also I use GH desktop instead of git so that's why I can't run the tests before commiting :(

You only need a command line. Non-GUI stuff was scary to me at first, too, but it's worth it :)

File: telegram/bot.py [hunk 1]
```diff
@@ -1170,20 +1183,21 @@ def edit_message_live_location(self,
...
             reply_markup (:class:`telegram.InlineKeyboardMarkup`, optional): A JSON-serialized
-                object for an inline keyboard.
+                object for a new inline keyboard.
             timeout (:obj:`int` | :obj:`float`, optional): If this value is specified, use it as
                 the read timeout from the server (instead of the one specified during creation of
                 the connection pool).
 
         Returns:
              :class:`telegram.Message`: On success the edited message.
+             :obj:`bool`: ``True`` is returned on failure.
```
Comment (Bibo-Joshi) [inline telegram/bot.py:1200]: `:obj:'True'` is returned, when the message to be edited was not sent by the bot. True wont be returned on failure ;)

File: telegram/bot.py [hunk 2]
```diff
@@ -1222,21 +1236,22 @@ def stop_message_live_location(self,
...
-                to remove reply keyboard or to force a reply from the user.
+            reply_markup (:class:`telegram.InlineKeyboardMarkup`, optional): A JSON-serialized
+                object for a new inline keyboard.
             timeout (:obj:`int` | :obj:`float`, optional): If this value is specified, use it as
                 the read timeout from the server (instead of the one specified during creation of
                 the connection pool).
 
         Returns:
             :class:`telegram.Message`: On success the edited message.
+            `:obj:`bool`: ``True`` is returned on failure.
```
Comment (Bibo-Joshi) [inline telegram/bot.py:1254]: see above

File: telegram/bot.py [hunk 3]
```diff
@@ -1454,7 +1469,8 @@ def send_chat_action(self, chat_id, action, timeout=None, **kwargs):
         """
         Use this method when you need to tell the user that something is happening on the bot's
         side. The status is set for 5 seconds or less (when a message arrives from your bot,
-        Telegram clients clear its typing status).
+        Telegram clients clear its typing status). We only recommend using this method when a
```
Comment (Bibo-Joshi) [inline telegram/bot.py:1472]: "Telegram only recommends …". We = PTB

File: telegram/bot.py [hunk 4]
```diff
@@ -3212,8 +3228,10 @@ def pin_chat_message(self, chat_id, message_id, disable_notification=None, timeo
     @log
     def unpin_chat_message(self, chat_id, timeout=None, **kwargs):
         """
-        Use this method to unpin a message in a supergroup. The bot must be an administrator in the
-        chat for this to work and must have the appropriate admin rights.
+        Use this method to unpin a message in a group, a supergroup, or a channel.
+        The bot must be an administrator in the chat for this to work and must have the
+        ‘can_pin_messages’ admin right in the supergroup or ‘can_edit_messages’ admin right
```
Comment (Bibo-Joshi) [inline telegram/bot.py:3233]: Please use "``" instead of "‘’". It renders better in sphinx

File: telegram/bot.py [hunk 5]
```diff
@@ -3596,9 +3619,8 @@ def stop_poll(self,
             chat_id (:obj:`int` | :obj:`str`): Unique identifier for the target chat or username
                 of the target channel (in the format @channelusername).
             message_id (:obj:`int`): Identifier of the original message with the poll.
-            reply_markup (:class:`telegram.ReplyMarkup`, optional): Additional interface options. A
-                JSON-serialized object for an inline keyboard, custom reply keyboard, instructions
-                to remove reply keyboard or to force a reply from the user.
+            reply_markup (:class:`telegram.InlineKeyboardMarkup`, optional): AA JSON-serialized
```
Comment (Bibo-Joshi) [inline telegram/bot.py:3622]: Single A

File: telegram/bot.py [hunk 6]
```diff
@@ -889,6 +899,8 @@ def send_animation(self,
                        **kwargs):
         """
         Use this method to send animation files (GIF or H.264/MPEG-4 AVC video without sound).
+        On success, the sent Message is returned. Bots can currently send animation files of
```
Comment (Bibo-Joshi) [inline telegram/bot.py:902]: Please use the "Returns:" thingy for the return type, as with the other functions. Can you change that for

* send_document
* send_sticker
* send_video
* send_video_note
* edit_message_media
* send_audio

And maybe also fix the rendering of the return description for stop_poll?

**Comment by harshil21** [comment]
> I did a bit of nitpicking ;)
> Please build the docs on your local machine to make sure that everything renders correctly. See the contrib-guide for details on that (you're gonna have to use the command line for that …)

Yea I finally set up this command line thing and installed the pre-commit hooks. I'll now use it to render the docs too.

File: telegram/bot.py (hunk 1 above)
Comment (harshil21) [reply telegram/bot.py:1200]: Oh right I'll change that

File: telegram/bot.py (hunk 4 above)
Comment (harshil21) [reply telegram/bot.py:3233]: Okay. I think I've seen those single quotes in other places too. I'll change all.

File: telegram/bot.py (hunk 6 above)
Comment (harshil21) [reply telegram/bot.py:902]: Yea I noticed that inconsistency. Wasn't sure which one to go for till now. Will change in all the other methods too.

> And maybe also fix the rendering of the return description for stop_poll?

Yea I noticed that a few minutes back while working on my pollbot. Will fix.

File: telegram/bot.py (hunk 3 above)
Comment (harshil21) [reply telegram/bot.py:1472]: I've seen 'We' being used in other places too where it should have been 'Telegram'. Changing it in those places too..

**Comment by Bibo-Joshi** [comment]
CI fails unrelated (this is docs only). merging.

**Pull request merged**